### PR TITLE
support for external config in BrooklynProperties

### PR DESCRIPTION
**Pull request by aledsage**
An alternative implementation to https://github.com/apache/incubator-brooklyn/pull/1018 (i.e. replaces that PR).

### CODE REVIEW

**Comment by aledsage** [comment]
Test failure is unrelated: `BindException: Address already in use` in `BrooklynApiRestClientTest.setUp`. This is caused by us finding a free port, and then immediately trying to use it - but another process running on the same jenkins test box has claimed that port.

We call `Networking.nextAvailablePort(FAVOURITE_PORT)` in `BrooklynRestApiLauncher.startServer`.  We should really let it choose a port, given that class is just in test code.

But that should be for a different PR.

Closing and re-opening to kick off jenkins again.

File: core/src/main/java/org/apache/brooklyn/core/mgmt/internal/DeferredBrooklynProperties.java [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+    public DeferredBrooklynProperties(BrooklynProperties delegate, ManagementContextInternal mgmt) {
+        this.delegate = checkNotNull(delegate, "delegate");
+        this.mgmt = checkNotNull(mgmt, "mgmt");
+    }
+    
+    private Object transform(ConfigKey<?> key, Object value) {
+        if (value instanceof CharSequence) {
+            String raw = value.toString();
+            if (raw.startsWith(EXTERNAL_CONFIG_PREFIX)) {
+                CampYamlParser parser = mgmt.getConfig().getConfig(CampYamlParser.YAML_PARSER_KEY);
```
Comment (ahgittin) [inline core/src/main/java/org/apache/brooklyn/core/mgmt/internal/DeferredBrooklynProperties.java]: why not `startsWith("$brooklyn:)` ?

**Comment by ahgittin** [comment]
nice work.  one substantive comment -- if not addressing, put a TODO in the doc.  but for instance i could see it useful to set an entity etc.  also pathological case but if user wanted to say

    external_prefix=$brooklyn:external

this would break it.  if we supported `literal` we could say:

    external_prefix=$brooklyn:literal("$brooklyn:external")

also docs need update:
* to say what is permitted in properties (ie currently only `$brooklyn:external(...)`)
* to say that the method for constructing `BrooklynProperties` has changed (release notes / api breakages)

**Comment by aledsage** [comment]
Thanks @ahgittin - great comments, and thanks for pointing out docs (too often ignored by me and others!). I've incorporated those changes and written some docs. Merging now.

**Pull request merged**